pkg/datapath/iptables/iptables.go

```diff
@@ -365,14 +365,14 @@ func (m *Manager) Start(ctx cell.HookContext) error {
 		m.disableIPEarlyDemux()
 	}
 
-	if err := m.modulesMgr.FindOrLoadModules(
+	if err := m.modulesMgr.FindOrLoadModules(m.sharedCfg.SkipKernelModuleValidation,
```

Comment:
Let's not make this a method which takes a boolean to "not do the thing". Like above for `m.disableIPEarlyDemux`, gate the calls to `FindOrLoadModules` with the configuration option. 

I know it's less DRY, but it's more readable. (I didn't check whether moving the tests over is doable, let's discuss further if it is not.)

Reply:
Maybe I'm overlooking the simple solution, but wouldn't this overcomplex the code quite a lot?

By doing this, I'd also need to move the `FindOrLoadModules` arguments (`"ip_tables", "iptable_nat", "iptable_mangle", "iptable_raw", "iptable_filter"`, `"ip6_tables", "ip6table_mangle", "ip6table_raw", "ip6table_filter"`, and `"xt_socket"`) each to a separate (global?) variable in `iptables.go`, as I'd still like this valuable information to be printed as part of the `skipped validating and loading the following kernel modules: %s)` info log line. Also, https://github.com/cilium/cilium/blob/95c8d7d3524ff5baf1cc2e2f4528baf6075a41e0/pkg/datapath/iptables/iptables.go#L387-L400 and https://github.com/cilium/cilium/blob/95c8d7d3524ff5baf1cc2e2f4528baf6075a41e0/pkg/datapath/iptables/iptables.go#L428 should still be run, when no `err` is returned from `FindOrLoadModules`, so I can't simply gate the full `if err := m.modulesMgr.FindOrLoadModules ... else ...` logic.